File: scripts/reference-solutions/first-bad-version.cpp
#include <iostream>
using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  long long n, b;
  cin >> n >> b;

  auto isBad = [&](long long version) { return version >= b; };

  long long lo = 1, hi = n;
  while (lo < hi) {
    long long mid = lo + (hi - lo) / 2;
    if (isBad(mid)) {
      hi = mid;
    } else {
      lo = mid + 1;
    }
  }
  cout << lo << '\n';
  return 0;
}

File: scripts/reference-solutions/first-bad-version.py
import sys


def main():
    data = sys.stdin.buffer.read().split()
    n = int(data[0])
    b = int(data[1])

    def is_bad(version):
        return version >= b

    lo, hi = 1, n
    while lo < hi:
        mid = lo + (hi - lo) // 2
        if is_bad(mid):
            hi = mid
        else:
            lo = mid + 1
    print(lo)


main()
